src/stlink-lib/chipid.c

```diff
@@ -1121,11 +1122,45 @@ void init_chipids(char *dir_to_scan)
       dump_a_chip (stderr, op);
       fprintf (stderr, "---------- new ------------\n");
       dump_a_chip (stderr, p);
-      
+
     }
   }
 #endif
 }
+#endif //STLINK_HAVE_DIRENT_H
+
+#if defined(_WIN32) && !defined(STLINK_HAVE_DIRENT_H)
+#include <fileapi.h>
+#include <strsafe.h>
+void init_chipids(char *dir_to_scan)
+{
+  HANDLE hFind = INVALID_HANDLE_VALUE;
+  WIN32_FIND_DATAA ffd;
+  char file_pattern[MAX_PATH] = {0};
+  char filepath[MAX_PATH] = {0};
+  StringCchCopyA(file_pattern, STLINK_ARRAY_SIZE(file_pattern), dir_to_scan);
+  if (FAILED(StringCchCatA(file_pattern, STLINK_ARRAY_SIZE(file_pattern), "\\*.chip"))) {
+    ELOG("Path to chips's dir too long.\n");
+    return;
+  };
+  hFind = FindFirstFileA(file_pattern, &ffd);
+  if (INVALID_HANDLE_VALUE == hFind){
```

Comment:
```suggestion
  if (INVALID_HANDLE_VALUE == hFind) {
```

Reply:
AFAIK this does not follow the current (intended) style. However the latter may not be completely consistent throughout the code.

Reply:
Is this about that (missing?) space before the opening parenthesis? It seems it’s done this way in this file in other `if`s and `for`s.